diff --git a/src/Core.js b/src/Core.js
index 3fcf439ab1f6d7092973838f9883b09335c1db57..592322f83e2e49677a688ff2e660c606ef1d482e 100644
--- a/src/Core.js
+++ b/src/Core.js
@@ -436,14 +436,23 @@ export default class Core {
 	 */
 	onPopstate = () => {
 		// don't trigger for on-page anchors
+		let windowLocationPathname = window.location.pathname;
+
+		if (window.location.pathname.slice(-1) === '/') {
+			windowLocationPathname = window.location.pathname.slice(0, -1);
+		}
+
 		if (
-			window.location.pathname === this.currentLocation.pathname
-			&& window.location.search === this.currentLocation.search
-			&& !this.isPopping
+			(
+				window.location.pathname === this.currentLocation.pathname &&
+				window.location.search === this.currentLocation.search &&
+				!this.isPopping
+			) || 
+			(this.currentLocation.pathname === windowLocationPathname)
 		) {
-			return false
+			return false;
 		}
-
+		
 		if (!this.allowInterruption && (this.isTransitioning || this.isPopping)) {
 			// overwrite history state with current page if currently navigating
 			window.history.pushState({}, '', this.popTarget)
